### ISSUE TITLE
[Core] Make Ray Core tasks/actors metrics counters (accumulators)

### ISSUE THREAD

**Issue by alexeykudinkin** [ray-project/ray#47522]
### What happened + What you expected to happen

Currently, a lot of critical Ray Core metrics like number `ray_tasks`, etc are being produced as gauges, therefore making it impossible to precisely track tasks state transitions. 

This happens b/c gauges are meant to be tracking point-in-time values and not incremental changes -- since gauge only stores the last value stored w/in the period it only shows snapshot of values of the last value written w/in the time-bucket instead of all incremental change

### Versions / Dependencies

2.35

### Reproduction script

NA

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by anyscalesam** [comment]
ACK - bring up for next planning cycle... cool @alexeykudinkin ?

**Comment by alexeykudinkin** [comment]
Yes, going to bump this to P0 after Ray Summit as this causing considerable confusion in troubleshooting due to inability to understand exactly state transition for specific tasks

**Comment by anyscalesam** [comment]
Added to JIRA

**Comment by alanwguo** [comment]
in addition, we should see if we can update the `instance` label to identify the instance the task or actor runs on instead of the instance that is emitting the metric. We should check the cardinality of doing such a change before implementing it but if possible, we can then have the ray tasks and actors graphs filter down to a particular instance

**Comment by alanwguo** [comment]
@MengjinYan , do you know if the `ray_tasks` instance label will be based on the node the task runs on today? 

I now for actors, the `instance` label is always the head node.

**Comment by MengjinYan** [comment]
From what I understand, the `instance` label is not attached by Ray. Instead, for actors and tasks, we attach `NodeAddress` tag on both coreworker and gcs:
https://github.com/ray-project/ray/blob/92946259efbcd2cb7388da3949480eef1b1371a1/src/ray/core_worker/core_worker_process.cc#L116
https://github.com/ray-project/ray/blob/92946259efbcd2cb7388da3949480eef1b1371a1/src/ray/gcs/gcs_server/gcs_server_main.cc#L86

This from what I understand, should be the address where the metric is generated

**Comment by MengjinYan** [comment]
@alexeykudinkin Going back to the original ask in the issue. If I understand correctly, the original concern was that the prometheus polling frequency is too low to reflect all state transition for all tasks. 

> * ray_tasks_total -- accumulator (counter) tracking all state transitions for every task (ie when task changes state we increment it's corresponding accumulator)

Wondering what's the granularity you are looking at for metrics (per task, per node, etc.)? And if per task information is needed, I'm wondering if the task event information would help here instead of new metrics?
